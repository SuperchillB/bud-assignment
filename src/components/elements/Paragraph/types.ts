import {
  BorderProps,
  ColorProps,
  FlexboxProps,
  LayoutProps,
  PositionProps,
  SpaceProps,
  TypographyProps,
} from 'styled-system';

export type ParagraphFunctionsProps = SpaceProps &
  ColorProps &
  LayoutProps &
  FlexboxProps &
  BorderProps &
  PositionProps &
  TypographyProps;

export type ParagraphProps = {
  variant: 'body' | 'subTitle';
} & ParagraphFunctionsProps;
