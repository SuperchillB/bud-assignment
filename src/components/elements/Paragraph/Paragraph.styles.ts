import styled from 'styled-components';
import {
  border,
  color,
  compose,
  flexbox,
  layout,
  position,
  space,
  typography,
  variant,
} from 'styled-system';
import { ParagraphProps } from './types';

const paragraphStyles = {
  mt: '0rem',
  mb: '0rem',
  fontWeight: 'normal',
};

export const Paragraph = styled.p<ParagraphProps>(
  variant({
    variants: {
      body: {
        ...paragraphStyles,
        fontSize: 'small',
      },
      subTitle: {
        ...paragraphStyles,
        fontSize: 'tiny',
      },
    },
  }),
  compose(space, color, layout, flexbox, border, position, typography),
);

Paragraph.defaultProps = {
  variant: 'body',
};
