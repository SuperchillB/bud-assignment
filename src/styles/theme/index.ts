import radii from './foundations/border';
import breakpoints from './foundations/breakpoints';
import { colors } from './foundations/colors';
import shadows from './foundations/shadows';
import space from './foundations/space';
import { fonts, fontSizes, fontWeights } from './foundations/typography';
import { ThemeType } from './types';

const theme: ThemeType = {
  breakpoints,
  colors,
  fonts,
  fontSizes,
  fontWeights,
  radii,
  shadows,
  space,
};

export default theme;
