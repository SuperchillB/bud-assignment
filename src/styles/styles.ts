import { createGlobalStyle } from 'styled-components';
import { normalize } from 'styled-normalize';
import { ThemeType } from './theme/types';

export const GlobalStyles = createGlobalStyle<{ theme: ThemeType }>`
  ${normalize};
  *, :before, :after {
    box-sizing: border-box;
    margin: 0;
    padding: 0;
  }
  html {
    font-size: 62.5%;
    scroll-behavior: smooth;
  }
  body {
    /* font-family: ${(props) => props.theme.fonts.primary}; */
    font-family: ${({ theme }) => theme.fonts.primary};
    font-size: 1.6rem;
    background: ${({ theme }) => theme.colors.info};
    color: ${(props) => props.theme.colors.primary}; // ! CHANGE LATER
    cursor: default;
  }
  main {
    height: 100%;
  }
  h1,h2,h3,h4,h5,h6,button {
    font-family: ${(props) => props.theme.fonts.title}; // ! CHANGE LATER
    line-height: 2rem;
  }
  p {
    line-height: 2rem;
  }
  h1 {
    font-weight: normal;
    font-size: 3.2rem;
  }
  a {
    color: inherit;
    text-decoration: none;
  }
  button {
    border: 0;
    padding: 0;
    background: none;
    cursor: pointer;
  }
  li{
    list-style: none;
  }
`;
