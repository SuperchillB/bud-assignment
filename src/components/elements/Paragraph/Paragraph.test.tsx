import { render, screen as screenRTL } from '@testing-library/react';
import { Paragraph } from './Paragraph.styles';

describe('Paragraph component', () => {
  test('renders its text and id', () => {
    const id = 'testId';
    const children = 'Test text';
    const { container } = render(
      <Paragraph id={id} variant="body">
        {children}
      </Paragraph>,
    );

    expect(container.querySelector('p')?.id).toEqual(id);
    expect(screenRTL.getByText(children)).toBeInTheDocument();
  });

  test('renders with given props and "body" variant', () => {
    const { container } = render(
      <Paragraph
        ml="medium"
        mt={['small', 'small', 'medium', 'large']}
        variant="body"
      />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders with given props and "subTitle" variant', () => {
    const { container } = render(
      <Paragraph
        ml="medium"
        mt={['small', 'small', 'medium', 'large']}
        variant="subTitle"
      />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
