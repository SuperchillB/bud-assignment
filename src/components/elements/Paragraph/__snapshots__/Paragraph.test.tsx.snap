// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph component renders with given props and "body" variant 1`] = `
<p
  class="sc-bczRLJ UJHAY"
/>
`;

exports[`Paragraph component renders with given props and "subTitle" variant 1`] = `
<p
  class="sc-bczRLJ kUyRJZ"
/>
`;
